### PR TITLE
Copter: merge ArduCopter.cpp and Copter.cpp

### PR DESCRIPTION
**Pull request by peterbarker**

### CODE REVIEW

**Comment by rmackay9** [comment]
Cool, thanks

File: ArduCopter/Copter.cpp [hunk 1]
```diff
@@ -20,6 +82,515 @@
...
+    nav_info.wp_bearing = flightmode->wp_bearing();
+    nav_info.wp_xtrack_error = flightmode->crosstrack_error() * 1.0e-2f;
+    nav_info.wp_number = mode_auto.mission.get_current_nav_index();
+    osd.set_nav_info(nav_info);
+}
+#endif
+
 /*
   constructor for main Copter class
  */
```
Comment (OXINARF) [inline ArduCopter/Copter.cpp]: Nitpicking: I would put the constructor as the first thing on the file.

File: ArduCopter/Copter.cpp [hunk 2]
```diff
@@ -12,6 +12,68 @@
...
+ *  Pat Hickey          :Hardware Abstraction Layer (HAL)
+ *  Robert Lefebvre     :Heli Support, Copter LEDs
+ *  Roberto Navoni      :Library testing, Porting to VRBrain
+ *  Sandro Benigno      :Camera support, MinimOSD
+ *  Sandro Tognana      :PosHold flight mode
+ *  Sebastian Quilter   :SmartRTL
+ *  ..and many more.
+ *
+ *  Code commit statistics can be found here: https://github.com/ArduPilot/ardupilot/graphs/contributors
+ *  Wiki: http://copter.ardupilot.org/
```
Comment (OXINARF) [inline ArduCopter/Copter.cpp]: Not holding this PR at all, but we should likely start a CONTRIBUTORS.md file, probably based on the allcontributors.org spec.

File: ArduCopter/Copter.cpp (hunk 1 above)
Comment (peterbarker) [reply ArduCopter/Copter.cpp]: I put it down where we created the instance of it on purpose.

@rmackay9 - preferences?

**Comment by rmackay9** [comment]
On the dev call we decided that this was a good change for all vehicles but it's up to the vehicle maintainers to decide when they should implement this in order to reduce havoc with outstanding PRs.

**Comment by rmackay9** [comment]
Merged, thanks!

**Pull request merged**